Geometry/GEMGeometry/src/GEMGeometry.cc

```diff
@@ -23,7 +23,7 @@ const GEMGeometry::DetIdContainer& GEMGeometry::detIds() const { return theDetId
 const GeomDet* GEMGeometry::idToDetUnit(DetId id) const { return dynamic_cast<const GeomDet*>(idToDet(id)); }
 
 const GeomDet* GEMGeometry::idToDet(DetId id) const {
-  mapIdToDet::const_iterator i = theMap.find(id);
+  mapIdToDet::const_iterator i = theMap.find(DetId(GEMDetId(id)));
```

Comment:
@bsunanda - I do not understand why do you need to downcast and then upcast it.

Reply:
There are parts of the code which refers to geometry by converting rawId to DetId - now there is a change in format for GEMDetId class. GEMDetId class knows that but not DetId. One possibility is to use a map of GEMDetId rather than DetId and then only down casting will be required.

Reply:
Please suggest how best we can address this?

Reply:
Trying only with down casting now.